### PR TITLE
Refactor `mesheryctl mesh` commands

### PR DESCRIPTION
**Pull request by alphaX86**
Signed-off-by: Aadhitya A <aadhitya864@gmail.com>

**Notes for Reviewers**

This PR is refactored version of #5375 

Thanks @jbyers19 for authoring the changes

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
This is an exciting one. :)

**Comment by alphaX86** [comment]
Checklist (so far done):
- ✅ deploy
- ✅ remove
- ⚠️ validate
- ⚠️ mesh (Base command)

File: mesheryctl/pkg/utils/helpers.go [hunk 1]
```diff
@@ -646,10 +646,11 @@ func GetSessionData(mctlCfg *config.MesheryCtlConfig) (*models.Preference, error
 	if err != nil {
 		return nil, err
 	}
+
 	prefs := &models.Preference{}
 	err = utils.Unmarshal(string(body), prefs)
 	if err != nil {
-		return nil, err
+		return nil, errors.New("Failed to process JSON data. Did you sign in Meshery?")
```
Comment (hexxdump) [inline mesheryctl/pkg/utils/helpers.go:653]:
```suggestion
		return nil, errors.New("Failed to process JSON data. Please sign into Meshery")
```

File: mesheryctl/internal/cli/root/mesh/remove.go [hunk 2]
```diff
@@ -17,7 +18,10 @@ var (
 		Long:  `remove service mesh in the connected kubernetes cluster`,
 		Example: `
 // Remove a service mesh
```
Comment (hexxdump) [inline mesheryctl/internal/cli/root/mesh/remove.go:20]:
```suggestion
// Remove a service mesh, linkerd
```

File: mesheryctl/internal/cli/root/mesh/remove.go [hunk 3]
```diff
@@ -17,7 +18,10 @@ var (
...
-mesheryctl mesh remove [mesh adapter name]
+mesheryctl mesh remove linkerd
+
+// Remove a service mesh under a specific namespace
```
Comment (hexxdump) [inline mesheryctl/internal/cli/root/mesh/remove.go:23]:
```suggestion
// Remove a service mesh(linkerd) under a specific namespace(linkerd-ns)
```

File: mesheryctl/internal/cli/root/mesh/mesh.go [hunk 4]
```diff
@@ -1,41 +1,292 @@
...
+			// verify the specified mesh is valid
+			// if no mesh was specified, the user will be prompted to select one
+			meshName, err = validateMesh(mctlCfg, meshName)
+			if err != nil {
+				log.Fatalln(err)
+			}
+
+			// ensure the mesh's adapter is available and update adapterURL if so
+			if err = validateAdapter(mctlCfg, meshName); err != nil {
+				// ErrValidatingAdapter
```
Comment (hexxdump) [inline mesheryctl/internal/cli/root/mesh/mesh.go:74]:
```suggestion
```

File: mesheryctl/internal/cli/root/mesh/mesh.go [hunk 5]
```diff
@@ -1,41 +1,292 @@
...
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return "", err
+	}
+	return string(body), nil
+}
+
+func sendValidateRequest(mctlCfg *config.MesheryCtlConfig, query string, delete bool) (string, error) {
+	path := mctlCfg.GetBaseMesheryURL() + "/api/system/adapter/operation"
```
[the remark this answers is not in the document]
Comment (alphaX86) [reply mesheryctl/internal/cli/root/mesh/mesh.go:190]: I'll check and update accordingly

**Comment by leecalcote** [comment]
@deepakdinesh1123, would you like to give this a review?

File: mesheryctl/internal/cli/root/mesh/mesh.go [hunk 6]
```diff
@@ -1,41 +1,310 @@
...
+		return "", ErrCreatingDeployResponseStream(err)
+	}
+
+	timer := time.NewTimer(time.Duration(1200) * time.Second)
+	eventChan := make(chan string)
+
+	//Run a goroutine to wait for the response
+	go func() {
+		for i := range event {
+			if strings.Contains(i.Data.Details, query) {
```
Comment (hexxdump) [inline mesheryctl/internal/cli/root/mesh/mesh.go:242]: can you point me to valid API response for any deployment(install/remove) request?

Comment (alphaX86) [reply mesheryctl/internal/cli/root/mesh/mesh.go:242]: The message after request won't be printed... still...

https://github.com/alphaX86/meshery/blob/9e69993e7ac0afcc3d604441182f8cce3e544bdc/mesheryctl/internal/cli/root/mesh/mesh.go#L152-L154

I refactored the above function to support install/remove/validate operations

Comment (hexxdump) [reply mesheryctl/internal/cli/root/mesh/mesh.go:242]: so for the deployment case, you might not have a query field in the payload, so why are you checking for it in the waitForDeployResponse() ?

Comment (alphaX86) [reply mesheryctl/internal/cli/root/mesh/mesh.go:242]: If --watch is applied

**Comment by Chadha93** [comment]
@alphaX86  Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

File: mesheryctl/internal/cli/root/mesh/mesh.go (hunk 6 above)
Comment (hexxdump) [reply mesheryctl/internal/cli/root/mesh/mesh.go:242]: got it

**Review by hexxdump — APPROVED** [review]
LGTM

**Pull request merged**